components/raftstore-v2/src/router/response_channel.rs

```diff
@@ -203,13 +208,16 @@ impl<'a, Res> Future for WaitResult<'a, Res> {
             };
             if check_bit(e, fired_bit).is_some() {
                 unsafe {
-                    return Poll::Ready((*self.core.res.get()).take());
+                    return Poll::Ready((*self.sub.core.res.get()).take());
                 }
             }
         }
     }
 }
 
+unsafe impl<'a, Res: Send> Send for WaitResult<'a, Res> {}
+unsafe impl<'a, Res: Send> Sync for WaitResult<'a, Res> {}
```

Comment:
No, it doesn't need to. `Sync` is guaranteed by the definition of `WaitResult`.

Reply:
It will catch warning no matter the implementation.

Reply:
Can you provide steps to trigger the warnings?

Reply:
I haven't run [Rudra](https://github.com/sslab-gatech/Rudra) personally, but in the past the security team has alerted a similar [pattern](https://github.com/tikv/raft-engine/pull/146) in raft-engine.

Reply:
The case you provided was probably the `Mutex` didn't seem to protect any data to Rudra as in Rust data is expected to put inside the `Mutex`. Here the situation is different. The implementation pattern is very common in lock-free data structs. I'm not sure whether Rudra can recognize it well, but if it doesn't then we may need to consider fixing the tools as it can also make a lot of false positive for many libraries in practice.

Reply:
Where is sync needed here? It seems all usage within this file can be changed to mutable reference.

Reply:
It's not needed any more as now `BaseSubscriber` is borrowed.